editor/src/components/canvas/canvas-strategies/strategies/drag-to-insert-metastrategy.tsx

```diff
@@ -19,58 +26,131 @@ import {
   insertElementInsertionSubject,
 } from '../../commands/insert-element-insertion-subject'
 import { updateFunctionCommand } from '../../commands/update-function-command'
-import { ParentBoundsForInsertion } from '../../controls/parent-bounds'
-import { ImmediateParentOutlines } from '../../controls/parent-outlines'
+import { ParentBounds } from '../../controls/parent-bounds'
+import { ParentOutlines } from '../../controls/parent-outlines'
 import { DragOutlineControl } from '../../controls/select-mode/drag-outline-control'
 import { FlexReparentTargetIndicator } from '../../controls/select-mode/flex-reparent-target-indicator'
 import {
-  findCanvasStrategy,
+  CanvasStrategyFactory,
+  MetaCanvasStrategy,
   pickCanvasStateFromEditorStateWithMetadata,
-  RegisteredCanvasStrategies,
 } from '../canvas-strategies'
 import {
-  InteractionCanvasState,
-  CustomStrategyState,
   CanvasStrategy,
-  getInsertionSubjectsFromInteractionTarget,
   controlWithProps,
-  strategyApplicationResult,
+  CustomStrategyState,
   emptyStrategyApplicationResult,
+  getInsertionSubjectsFromInteractionTarget,
+  InteractionCanvasState,
   InteractionLifecycle,
+  strategyApplicationResult,
   targetPaths,
 } from '../canvas-strategy-types'
-import { InteractionSession } from '../interaction-state'
+import { InteractionSession, MissingBoundsHandling } from '../interaction-state'
+import { getApplicableReparentFactories } from './reparent-metastrategy'
+import { ReparentStrategy } from './reparent-strategy-helpers'
 
-export function dragToInsertStrategy(
+export const dragToInsertMetaStrategy: MetaCanvasStrategy = (
   canvasState: InteractionCanvasState,
   interactionSession: InteractionSession | null,
   customStrategyState: CustomStrategyState,
-): CanvasStrategy | null {
+): Array<CanvasStrategy> => {
+  if (
+    interactionSession == null ||
+    !(
+      interactionSession.interactionData.type === 'DRAG' ||
+      interactionSession.interactionData.type === 'HOVER'
+    )
+  ) {
+    return []
+  }
+
+  const { interactionData } = interactionSession
+
   const insertionSubjects = getInsertionSubjectsFromInteractionTarget(canvasState.interactionTarget)
   if (insertionSubjects.length === 0) {
-    return null
+    return []
   }
 
+  const pointOnCanvas =
+    interactionData.type === 'DRAG'
+      ? offsetPoint(interactionData.originalDragStart, interactionData.drag ?? zeroCanvasPoint)
+      : interactionData.point
+
+  const cmdPressed = interactionData.modifiers.cmd
+
+  const applicableReparentFactories = getApplicableReparentFactories(
+    canvasState,
+    pointOnCanvas,
+    cmdPressed,
+    true, // <- TODO this is an important assumption, make sure this is still true when inserting into a flex storyboard
```

Comment:
Later work?

Reply:
Yeah. This is a duplicate `TODO` from the draw-to-insert-metastrategy, as both need to be addressed together. I'll make sure it's tracked in the doc actually